community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NumberArrayFactory.java

```diff
@@ -179,6 +245,24 @@ public IntArray newIntArray( long length, int defaultValue )
             throw error( length, 4, error );
         }
 
+        @Override
+        public ByteArray newByteArray( long length, byte[] defaultValue, long base )
+        {
+            OutOfMemoryError error = null;
+            for ( NumberArrayFactory candidate : candidates )
+            {
+                try
+                {
+                    return candidate.newByteArray( length, defaultValue, base );
+                }
+                catch ( OutOfMemoryError e )
```

Comment:
why do we need to catch OOM here and try to allocate next one if we already just hit the limit?